src/ring_theory/localization.lean

```diff
@@ -1390,6 +1393,96 @@ def localization_localization_at_prime_iso_localization (p : ideal (localization
+
+/-- If `M ≤ N` are submonoids of `R`, then `N⁻¹S` is also the localization of `M⁻¹S` at `N`. -/
+lemma is_localization_of_submonoid_le
+  (M N : submonoid R) (h : M ≤ N) [is_localization M S] [is_localization N T]
+  [algebra S T] [is_scalar_tower R S T] :
+  is_localization (N.map (algebra_map R S).to_monoid_hom) T :=
+begin
+  constructor,
+  { rintro ⟨_, ⟨y, hy, rfl⟩⟩,
+    convert is_localization.map_units T ⟨y, hy⟩,
+    exact (is_scalar_tower.algebra_map_apply _ _ _ _).symm },
+  { intro y,
+    obtain ⟨⟨x, s⟩, e⟩ := is_localization.surj N y,
+    refine ⟨⟨algebra_map _ _ x, _, _, s.prop, rfl⟩, _⟩,
+    simpa [← is_scalar_tower.algebra_map_apply] using e },
+  { intros x₁ x₂,
+    obtain ⟨⟨y₁, s₁⟩, e₁⟩ := is_localization.surj M x₁,
+    obtain ⟨⟨y₂, s₂⟩, e₂⟩ := is_localization.surj M x₂,
+    refine iff.trans _ (set.exist_image_iff (algebra_map R S) N (λ c, x₁ * c = x₂ * c)).symm,
+    dsimp only at e₁ e₂ ⊢,
+    suffices : algebra_map R T (y₁ * s₂) = algebra_map R T (y₂ * s₁) ↔
+      ∃ (a : N), algebra_map R S (a * (y₁ * s₂)) = algebra_map R S (a * (y₂ * s₁)),
+    { have h₁ := (is_localization.map_units T ⟨_, h s₁.prop⟩).mul_left_inj,
+      have h₂ := (is_localization.map_units T ⟨_, h s₂.prop⟩).mul_left_inj,
+      simp only [is_scalar_tower.algebra_map_apply R S T, subtype.coe_mk] at h₁ h₂,
+      simp only [is_scalar_tower.algebra_map_apply R S T, map_mul, ← e₁, ← e₂, ← mul_assoc,
+        mul_right_comm _ (algebra_map R S s₂),
+        mul_right_comm _ (algebra_map S T (algebra_map R S s₂)),
+        (is_localization.map_units S s₁).mul_left_inj,
+        (is_localization.map_units S s₂).mul_left_inj] at this,
+      rw [h₂, h₁] at this,
+      simpa only [mul_comm] using this },
+    simp_rw [is_localization.eq_iff_exists N T, is_localization.eq_iff_exists M S],
+    split,
+    { rintro ⟨a, e⟩, exact ⟨a, 1, by { convert e using 1; simp; ring }⟩ },
+    { rintro ⟨a, b, e⟩, exact ⟨a * (⟨_, h b.prop⟩ : N), by { convert e using 1; simp; ring }⟩ } }
+end
+
+/-- If `M ≤ N` are submonoids of `R` such that `∀ x : N, ∃ m : R, m * x ∈ M`, then the
+localization at `N` is equal to the localizaton of `M`. -/
+lemma is_localization_of_is_exists_mul_mem
+  {R : Type*} [comm_ring R] (M : submonoid R) (S : Type*)
+    [comm_ring S] [algebra R S] [is_localization M S] (N : submonoid R) (h : M ≤ N)
+    (h' : ∀ x : N, ∃ m : R, m * x ∈ M) : is_localization N S :=
+begin
+  constructor,
+  { intro y,
+    obtain ⟨m, hm⟩ := h' y,
+    have := is_localization.map_units S ⟨_, hm⟩,
+    erw map_mul at this,
+    exact (is_unit.mul_iff.mp this).2 },
+  { intro z,
+    obtain ⟨⟨y, s⟩, e⟩ := is_localization.surj M z,
+    exact ⟨⟨y, _, h s.prop⟩, e⟩ },
+  { intros x₁ x₂,
+    rw is_localization.eq_iff_exists M,
+    refine ⟨λ ⟨x, hx⟩, ⟨⟨_, h x.prop⟩, hx⟩, _⟩,
+    rintros ⟨x, h⟩,
+    obtain ⟨m, hm⟩ := h' x,
+    refine ⟨⟨_, hm⟩, _⟩,
+    simp [mul_comm m, ← mul_assoc, h] }
+end
```

Comment:
Please rewrite this using record syntax. It should fit on the same number of lines, but it also gives the field names explicitly, which improves readability a bit.